tests/ui/assertions_on_constants.rs

```diff
@@ -18,6 +18,8 @@ fn main() {
     assert!(C);
 
     debug_assert!(true);
+    // Don't lint this, since there is no better way for expressing "Only panic in debug mode".
+    debug_assert!(false); // #3948
```

Comment:
It would be good to add a test case for the `debug_assert_not_in_macro` check as well.

Reply:
It's already in the `assert_const!` macro in this test file.
```rust
macro_rules! assert_const {
    ($len:expr) => {
        assert!($len > 0);
        debug_assert!($len < 0);
    };
}
```

Reply:
Oh, alright!